### PR TITLE
feat(UI): environment group actions

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**
Collapsed group actions:
Screenshot 2024-09-23 at 16 35 56

Expanded group actions:
Screenshot 2024-09-25 at 08 54 26

Ref: SRX-HYVGNP

### CODE REVIEW

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 1]
```diff
@@ -445,10 +446,82 @@ export const EnvironmentGroupLane: React.FC<{
...
+            return;
+        }
+
+        if (allReleases.length === 0) {
+            setAllowDeployment(true);
+        } else {
+            setAllowDeployment(
+                JSON.stringify(allReleases[0].environments) !== JSON.stringify(environmentGroup.environments)
+            );
+        }
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:522]: this seems more complicated than necessary, or maybe I don't understand the goal. Why is it ever necessary to disable the expand button?
But assuming it is necessary:
Please rename "allowDeployment" to something like "enableExpandGroup".

Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:522]: Oh, we are not talking about "the expand button" but the deployment button for a group.
In that case maybe "allowGroupDeployment" as a name?

File: services/frontend-service/src/ui/components/button/ExpandButton.tsx [hunk 2]
```diff
@@ -97,3 +97,93 @@ export const ExpandButton = (props: ExpandButtonProps): JSX.Element => {
...
 };
+
+export type EnvGroupExpandButtonProps = {
+    onClickSubmit: (shouldLockToo: boolean) => void;
+    onClickLock: () => void;
+    defaultButtonLabel: string;
+    disabled: boolean;
+};
+
+export const EnvGroupExpandButton = (props: EnvGroupExpandButtonProps): JSX.Element => {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/button/ExpandButton.tsx:108]: This one needs some tests.

Comment (miguel-crespo-fdc) [reply services/frontend-service/src/ui/components/button/ExpandButton.tsx]: Merged with ExpandButton. Added one test case to test clicking it

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**